src/Plugin/Field/FieldFormatter/VideoFormatter.php

```diff
@@ -0,0 +1,230 @@
+      $container->get('media.oembed.url_resolver'),
+      $container->get('http_client'),
+      $container->get('path.validator'),
+      $container->get('config.factory')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @throws \GuzzleHttp\Exception\GuzzleException
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode): array {
+    $elements['#attached']['library'][] = 'itk_video/video';
+
+    foreach ($items as $delta => $item) {
+
+      $this->createVideo($item);
+
+      if (!empty($item->getUrl()->toString())) {
+        $markup =
+          '<div class="itk-video itk-video-responsive">' . $this->createVideo($item) . '</div>';
+
+        $elements[$delta] = [
+          '#type' => 'inline_template',
+          '#template' => $markup,
+        ];
+      }
+    }
+
+    return $elements;
+  }
+
+  /**
+   * Render a video from an embed url or iframe.
+   *
+   * @param \Drupal\itk_video\Plugin\Field\FieldType\Video $value
+   *   The video field type.
+   *
+   * @return string|null
+   *   The rendered html.
+   *
+   * @throws \GuzzleHttp\Exception\GuzzleException
+   *   An exception if guzzle fails.
+   */
+  private function createVideo(Video $value): ?string {
+    $settings = $this->configFactory->get('itk_video.settings');
+
+    // Set string.
+    $url = $value->getUrl()->toString();
+
+    $videoArray = $this->createVideoFromUrl($url, $settings);
+
+    if ($settings->get('respect_cookie_information')) {
+      $videoArray = $this->applyCookieConsent($videoArray, $value);
+    }
+
+    return $videoArray['iframe'] ?? NULL;
+  }
+
+  /**
+   * Create an array containing video information.
+   *
+   * @param string $text
+   *   The text input to create video from.
+   * @param \Drupal\Core\Config\ImmutableConfig $settings
+   *   The settings for the field.
+   *
+   * @return array
+   *   The resulting video array.
+   *
+   * @throws \GuzzleHttp\Exception\GuzzleException
+   *   Exception if oembed fails.
+   */
+  private function createVideoFromUrl(string $text, ImmutableConfig $settings): array {
+    $video = [];
+    if (filter_var($text, FILTER_VALIDATE_URL)) {
+      $supportedProviders = SupportedVideoProviders::getConfig();
+      $providersStatus = $settings->get('providers_status');
+
+      $url = parse_url($text);
+      if (in_array($url['host'], SupportedVideoProviders::getProviderHosts())) {
+        $video['host'] = $url['host'];
+
+        $providerKey = $this->getProviderIdFromHost($supportedProviders, $video['host']);
+
+        if (!empty($providerKey && $providersStatus[$providerKey])) {
+          // Use oembed to create iframe if possible.
+          if ('Oembed' === $supportedProviders[$providerKey]['type']) {
+            try {
+              $url = $this->urlResolver->getResourceUrl($text);
+              $request = $this->httpClient->request('GET', $url);
+              $status = $request->getStatusCode();
+              if (200 == $status) {
```

Comment:
```suggestion
              if (Response::HTTP_OK === $status) {
```

(and `use Symfony\Component\HttpFoundation\Response`).